File: python/eduid.py
# -*- coding: utf-8 -*-
import uuid

def GenerateShanghaiEduID(IdNumber):
    ShanghaiEduDomain = "sh.edu.cn"
    eduidNameSpace = uuid.uuid5(uuid.NAMESPACE_DNS, ShanghaiEduDomain)
    return uuid.uuid5(eduidNameSpace, IdNumber)
    
if __name__ == '__main__':
    print(GenerateShanghaiEduID("110101200101016874"))

File: golang/eduid.go
package eduid

import (
	"github.com/shanghai-edu/go.uuid"
)

const (
	ShanghaiEduDomain = "sh.edu.cn"
)

func GenerateShanghaiEduID(IdNumber string) (uuidString string) {
	ns := uuid.NewV5(uuid.NamespaceDNS, ShanghaiEduDomain)
	u5 := uuid.NewV5(ns, IdNumber)
	uuidString = u5.String()
	return
}
